test/Test/Database/LSMTree/StateMachine.hs

```diff
@@ -1900,6 +1983,9 @@ arbitraryActionWithVars _ label ctx (ModelState st _stats) =
           , (1, R.UnionCredits <$> QC.arbitrary)
           ]
 
+        -- TODO: tweak distribution once table unions are implemented
+        genPortion = Portion <$> QC.elements [1, 2, 3]
```

Comment:
Ah, nice, so no matter much much debt, we always have a chance to finish the union immediately.